src/Libplanet.Types/Assets/Currency.cs

```diff
@@ -700,18 +438,130 @@ public IValue Serialize()
                 serialized = serialized.Add("totalSupplyTrackable", true);
                 if (MaximumSupply is { } maximumSupply)
                 {
-                    serialized = serialized.Add(
-                        "maximumSupplyMajor",
-                        (IValue)new Integer(maximumSupply.MajorUnit)
-                    ).Add(
-                        "maximumSupplyMinor",
-                        (IValue)new Integer(maximumSupply.MinorUnit));
+                    serialized = serialized
+                        .Add("maximumSupplyMajor", new Integer(maximumSupply.MajorUnit))
+                        .Add("maximumSupplyMinor", new Integer(maximumSupply.MinorUnit));
                 }
             }
 
             return serialized;
         }
 
+        private static string ExtractTicker(Dictionary dict) =>
+            ((Text)dict["ticker"]).Value;
+
+        private static byte ExtractDecimalPlaces(Dictionary dict)
+        {
+            if (!(dict.ContainsKey("decimals") ^ dict.ContainsKey("decimalPlaces")))
```

Comment:
Is the operator `^` the same as `||`? I've never used it like this before, so it's a bit awkward.